### PR TITLE
Add Link component

### PR DESCRIPTION
**Pull request by Joozty**
Closes https://github.com/oacore/design/issues/205

### CODE REVIEW

File: src/elements/link/index.js [hunk 1]
```diff
@@ -0,0 +1 @@
+export default from './link'
```
Comment (viktor-yakubiv) [inline src/elements/link/index.js:1]: I do not like this convention. Do you see any other better option to handle component modules?

File: src/elements/link/link.jsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+const Link = React.forwardRef(
+  ({ children, className, external = false, ...restProps }, ref) => (
+    <a
+      ref={ref}
+      className={classNames.use(styles.link).join(className)}
+      {...restProps}
+    >
+      {children}
+      {external && <Icon src="#iconhere" alt="external" />}
```
Comment (viktor-yakubiv) [inline src/elements/link/link.jsx:17]: The reason for `external` is not in the Icon, it's fine without, I would rather drop it. The prop is a shortcut for `target="_blank" rel="noopener noreferrer"`

Comment (Joozty) [reply src/elements/link/link.jsx:17]: alt is required prop. I just put something in there.  So you are saying we should drop required alt prop?

File: src/elements/link/index.js (hunk 1 above)
Comment (Joozty) [reply src/elements/link/index.js:1]: Not at the moment. :(

File: src/elements/link/link.jsx (hunk 2 above)
Comment (viktor-yakubiv) [reply src/elements/link/link.jsx:17]: I do, we should drop the requirement from the `Icon` but it's slightly different, out of this scope.

In the previous comment, I mean that the `external` prop should add `target="_blank"` but the icon itself is optional.

Comment (Joozty) [reply src/elements/link/link.jsx:17]: Thanks will add it. It still work in progress so I just put blindly things here.

**Comment by Joozty** [comment]
@viktor-yakubiv Don't you know by any chance the name of the external icon? It's like finding a needle in the haystack.

**Comment by viktor-yakubiv** [comment]
If I remember, `open-in-new`

**Comment by viktor-yakubiv** [comment]
> If user click on Link in menu they are redirected to the button link instead because it has the same ID.

The backside of the fancy markdown 😄

**Comment by viktor-yakubiv** [comment]
I am looking into it. I will send an update, I am exploring more cases. However, I like the overlall result. Thanks a lot for doing this 👍

**Comment by viktor-yakubiv** [comment]
I found the icon looks good in Safari but in Chrome and Firefox it does not look very good in text. Feel free to check it now but I will apply some changes.

By the way, I do not recommend to change the default display for the text elements, it may be unexpected for the user.

**Comment by viktor-yakubiv** [comment]
There is a problem that I see can only be solved with web-fonts at the moment (without JS processing). I wish I am wrong and there is an option with the background + mask bit obviously inline-block behaves like an image too.

Open in new icon falls to the next line

The icon should break with the last work within the link alway. `white-space` does not help us because it will make the whole link unbreakable.

**Comment by viktor-yakubiv** [comment]
I added a simple solution but we should investigate this breaking rules more because it will become a headache when goes to production.

**Comment by viktor-yakubiv** [comment]
@Joozty sorry I put my hands here maybe too much. I tried to explain my concerns as much as possible and even created an issue after this (#216). I wonder if you have any comments but if you don't feel free to rebase and merge 😉

Thanks a lot for looking into this. It was interesting investigation, I hope for both of us 🙂

**Comment by Joozty** [comment]
> The icon should break with the last work within the link alway. white-space does not help us because it will make the whole link unbreakable.

You would need another wrapper I guess.
```
<a>Link with the last <span>word<svg>icon here</svg></span></a>
```

Anyway thanks for looking into this. As I always say I don't see any difference whether the element is inline flex or just inline. 😄

**Comment by viktor-yakubiv** [comment]
This is exactly the solution I was thinking about but it needs language processing on JS what I would like to avoid. I would rather have this defect than handle it in such way.

**Pull request merged**